store/sqlstore/preference_store.go

```diff
@@ -282,45 +282,42 @@ func (s SqlPreferenceStore) DeleteCategoryAndName(category string, name string)
 	})
 }
 
-func (s SqlPreferenceStore) CleanupFlagsBatch(limit int64) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		query :=
-			`DELETE FROM
-				Preferences
-			WHERE
-				Category = :Category
-				AND Name IN (
+func (s SqlPreferenceStore) CleanupFlagsBatch(limit int64) (int64, *model.AppError) {
+	query :=
+		`DELETE FROM
+			Preferences
+		WHERE
+			Category = :Category
+			AND Name IN (
+				SELECT
+					*
+				FROM (
 					SELECT
-						*
-					FROM (
-						SELECT
-							Preferences.Name
-						FROM
-							Preferences
-						LEFT JOIN
-							Posts
-						ON
-							Preferences.Name = Posts.Id
-						WHERE
-							Preferences.Category = :Category
-							AND Posts.Id IS null
-						LIMIT
-							:Limit
-					)
-					AS t
-				)`
-
-		sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
-		if err != nil {
-			result.Err = model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+						Preferences.Name
+					FROM
+						Preferences
+					LEFT JOIN
+						Posts
+					ON
+						Preferences.Name = Posts.Id
+					WHERE
+						Preferences.Category = :Category
+						AND Posts.Id IS null
+					LIMIT
+						:Limit
+				)
+				AS t
+			)`
+
+	sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
+	if err != nil {
+		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+	} else {
+		rowsAffected, err1 := sqlResult.RowsAffected()
+		if err1 != nil {
+			return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
 		} else {
```

Comment:
Here you can also drop the `else`.

Reply:
Hi @hanzei , glad to help and thanks for bearing with me - it's my first contribution to open source (even though it was mentioned that this is for enterprise).

For the naming, you mean err1 can just be err, right? All that stuff I just kept as it was with the previous commit, but yes, I'll change it - I assume the following is fine?
```go

	sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
	if err != nil {
		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
	}

	rowsAffected, err := sqlResult.RowsAffected()
	if err != nil {
		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
	}

	return rowsAffected, nil
```

Reply:
That looks pretty good! :+1: